### PR TITLE
add license and contributing guide

### PR DESCRIPTION
**Pull request by blankdots**
- add License
- add Contributions guide
- refactor main readme to include info about API

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+If you're reading this, it means you saw something that is not right, you want to add a new feature or your manager asked you to contribute to this. In any case we are glad and it would be awesome if you can contribute.
+
+### Submitting Issues
+
+We have templates for submitting new issues, that you can fill out. For example if you found a bug, use the Bug Report template
+
+### Submitting changes
+
+When you made some changes you are happy with please send a [Github Pull Request to swift-browser-ui](https://github.com/CSCfi/sda-filesystem/pull_requests/new/master) with a clear list of what you've done (read more about [pull requests]https://help.github.com/en/articles/about-pull-requests)). When you create that Merge Request, we will forever be in your debt if you check previous docs for typos.
```
Comment (teemukataja) [inline CONTRIBUTING.md:11]:
```suggestion
When you made some changes you are happy with please send a [Github Pull Request to sda-filesystem](https://github.com/CSCfi/sda-filesystem/pull_requests/new/master) with a clear list of what you've done (read more about [pull requests]https://help.github.com/en/articles/about-pull-requests)). When you create that Merge Request, we will forever be in your debt if you check previous docs for typos.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   * `docs/<what-the-docs>` - missing docs or keeping them up to date;
+   * `bugfix/<caught-it>` - solved a bug;
+   * `test/<thank-you>` - adding missing tests for a feature, we would prefer they would come with the `feature` but still `thank you`;
+   * `refactor/<that-name-is-confusing>` - well we hope we don't mess anything and we don't get to use this;
+   * `hotfix/<oh-no>` - for when things needed to be fixed yesterday.
+
+
+### Coding conventions
+
+We do optimize for readability, and it would be awesome if you go through the code and see what conventions we've used so far.
```
Comment (teemukataja) [inline CONTRIBUTING.md:33]: Should we add how to run lint?
```
golangci-lint run --out-format=github-actions -E gosec,goconst,nestif,interfacer,bodyclose,rowserrcheck
```

File: README.md [hunk 3]
```diff
@@ -8,7 +12,10 @@ For test environment use:
 ```
 export SD_CONNECT_METADATA_API=https://connect-metadata-api-test.sd.csc.fi
 export SD_CONNECT_DATA_API=https://connect-data-api-test.sd.csc.fi
-export SD_CONNECT_CERTS=cert.pem	#SD_CONNECT_CERTS should be the file that contains the necessary certificates
+
+# Connection requires a certificate
```
Comment (teemukataja) [inline README.md:16]: only if using untrusted (e.g. self-signed) certificates, is signed by a trusted CA, this is not needed

File: CONTRIBUTING.md (hunk 2 above)
Comment (blankdots) [reply CONTRIBUTING.md:33]: done with #3

**Pull request merged**